config/gulp/file-upload.js

```diff
@@ -1,69 +1,51 @@
 require("dotenv").config();
-const { series, src } = require("gulp"),
-  vinylPaths = require("vinyl-paths"),
-  del = require("del"),
-  s3config = {
-    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+const gulp = require("gulp");
+const awspublish = require('gulp-awspublish');
+const del = require("del");
+const vinylPaths = require("vinyl-paths");
+
+// Create a new publisher using S3 options
+const publisher = awspublish.create({
+ //region: 'your-region-id', // Change to your AWS region
+  params: {
+    Bucket: 'digitalgov', // Change to your bucket name
   },
-  s3 = require("gulp-s3-upload")(s3config);
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+});
 
-/**
- * Uploads images in /processed folder to digitalgov s3 bucket
- * TODO: Refactor both functions into one
- * https://github.com/uswds/uswds-compile/blob/223fe67c335135c55cf1be9dead44e6d363e219d/gulpfile.js#L204-L232
- */
+// Define custom headers
+const headers = {
+  'Cache-Control': 'max-age=315360000, no-transform, public',
+};
 
 function uploadImage() {
   console.log("starting image upload");
-
-  return src("content/uploads/_working-images/processed/*")
-    .pipe(
-      s3(
-        {
-          Bucket: "digitalgov",
-          ACL: "public-read",
-        },
-        {
-          maxRetries: 5,
-        }
-      )
-    )
+  return gulp.src("content/uploads/_working-images/processed/*")
+    .pipe(awspublish.gzip({ ext: '.gz' }))
+    .pipe(publisher.publish(headers))
+    .pipe(publisher.cache())
+    .pipe(awspublish.reporter())
     .pipe(vinylPaths(del));
 }
 
-/**
- * Uploads files in /to-process folder to s3 static file bucket
- * TODO: Refactor both functions into one
- */
-
 function uploadFile() {
   console.log("starting file upload");
-
-  return src("content/uploads/_working-files/to-process/*")
-    .pipe(
-      s3(
-        {
-          Bucket: "digitalgov/static",
-          ACL: "public-read",
-        },
-        {
-          maxRetries: 5,
-        }
-      )
-    )
+  publisher.config.params.Bucket = "digitalgov/static"; // Change to your specific static files bucket
```

Comment:
I get this error when running `npx gulp watch`:

```bash
[10:38:52] InvalidBucketName: Bucket name shouldn't contain '/', received 'digitalgov/static'
```

Might need to look here for AWS 3 changes — https://docs.aws.amazon.com/AmazonS3/latest/userguide/BucketRestrictions.html?

Reply:
I believe this is solved please verify